warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -263,6 +258,32 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
         if (env != null) {
             ActiveQueryLog.setConfig(env.getConfig());
         }
+        
+        pruneIvaratorCacheBaseURIs();
+    }
+    
+    public void pruneIvaratorCacheBaseURIs() {
+        if (ivaratorCacheBaseURIAlternatives != null) {
+            List<String> newAlternatives = new ArrayList<>(ivaratorCacheBaseURIAlternatives);
```

Comment:
Seems like it should be needed, but the list implementation in `ivaratorCacheBaseURIAlternatives` doesn't support `remove`.

Reply:
That's whack yo!  But I get it!